lib/game.rb

```diff
@@ -0,0 +1,76 @@
+require 'pry'
+
+class Game
+    
+    WIN_COMBINATIONS = [
+        [0,1,2],
+        [3,4,5],
+        [6,7,8],
+        [0,3,6],
+        [1,4,7],
+        [2,5,8],
+        [0,4,8],
+        [2,4,6]
+    ]
+    
+    attr_accessor :board, :player_1, :player_2
+    
+    def initialize(player_1 = Player::Human.new("X"), player_2 = Player::Human.new("O"), board = Board.new)
+        @board = board
+        @player_1 = player_1
+        @player_2 = player_2
+    end
+    
+    def current_player
+        board.turn_count.odd? ? player_2 : player_1 
+    end
+    
+    def over?
+      won? || draw?
+    end
+    
+    def won?
+      WIN_COMBINATIONS.each do |array|
+        if array.all? { |item| board.cells[item] == 'X'} || 
+          array.all? { |item| board.cells[item] == 'O'}
+          return array
+        end
+      end
+     false
+    end
+    
+    def draw?
+        board.full? && !won? 
+    end
+    
+    def winner
+      board.cells[won?[0]] if won?
```

Comment:
Instead of calling `Game#won?` twice, consider using an if statement assignment. For example:

```ruby
if won = won?
  board.cells[won[0]]
end
```

If `won?` returns true, then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.